src/components/Waveform/Waveform.tsx

```diff
@@ -416,7 +417,17 @@ export const Waveform = forwardRef<IWaveformRef, IWaveform>((props, ref) => {
       result => {
         if (mode === 'live') {
           if (!isNil(result.currentDecibel)) {
-            setWaveform(prev => [...prev, result.currentDecibel]);
+            setWaveform((previousWaveform: number[]) => {
+              const updatedWaveform: number[] = [
+                ...previousWaveform,
+                result.currentDecibel,
+              ];
+
+              if (updatedWaveform.length > maxCandlesToRender)
+                updatedWaveform.shift();
+
+              return updatedWaveform;
+            });
```

Comment:
We can use `slice` instead of `shift` as it is more performant.
```suggestion
            setWaveform((previousWaveform: number[]) => {
  // Add the new decibel to the waveform
  const updatedWaveform = [...previousWaveform, result.currentDecibel];

  // Limit the size of the waveform array to 'maxCandlesToRender'
  return updatedWaveform.length > maxCandlesToRender
    ? updatedWaveform.slice(1)
    : updatedWaveform;
});
```